### PR TITLE
Add support for Arc

### PR DESCRIPTION
**Pull request by ambv**
We expose pulse width, filter cutoff, filter resonance, and envelope release.

Additionally, Grid and Arc rotation is now supported via PSET params.

### CODE REVIEW

File: awake.lua [hunk 1]
```diff
@@ -299,34 +299,52 @@ function init()
   params:add{type = "trigger", id = "reset", name = "reset",
     action = function() reset() end}
   
-  params:add_group("synth",6)
+  params:add_group("synth", 7)
```
Comment (ambv) [inline awake.lua:302]: When Ezra added detune, it was supposed to be part of the synth> group but this wasn't changed, and so it landed in the top menu. This fixed it.

File: awake.lua [hunk 2]
```diff
@@ -299,34 +299,52 @@ function init()
...
   cs_AMP = controlspec.new(0,1,'lin',0,0.5,'')
   params:add{type="control",id="amp",controlspec=cs_AMP,
     action=function(x) engine.amp(x) end}
 
   cs_PW = controlspec.new(0,100,'lin',0,50,'%')
   params:add{type="control",id="pw",controlspec=cs_PW,
-    action=function(x) engine.pw(x/100) end}
+    action=function(x) engine.pw(x/100); arcredraw() end}
```
Comment (ambv) [inline awake.lua:309]: The redraw in the `step` function is pretty sparse, it goes as fast (or as slow) as the musical clock. If you change those params via Norns UI or via OSC, this looked bad as it was refreshing too rarely.

Putting the redraw here makes sure that any param:set and param:delta is immediately shown on the device, which looks entirely smooth. And thanks to putting this in the action, we don't have to worry about covering the bases manually: params set via UI, via Arc encoder, and via OSC are all refreshing Arc.

File: awake.lua [hunk 3]
```diff
@@ -360,6 +378,10 @@ function g.key(x, y, z)
 end
 
 function gridredraw()
+  if not g then
+    return
+  end
+
```
Comment (ambv) [inline awake.lua:384]: For consistency with `arcredraw()`.

File: awake.lua [hunk 4]
```diff
@@ -387,6 +409,67 @@ function gridredraw()
...
+  -- Arc doesn't have built-in rotation support so we do
+  -- it ourselves.
+  local rot = 16 * (params:get("arc_rotation") - 1)
+  a:all(0)
+  a:led(1, rot + 1, 4)
+  a:led(2, rot + 1, 4)
+  a:led(3, rot + 1, 4)
+  a:led(4, rot + 1, 4)
```
Comment (ambv) [inline awake.lua:440]: We show the top tick if no value is present. This lets the user see what Arc rotation is set.

**Comment by ambv** [comment]
Demo in `norns study group`: https://discord.com/channels/765746584582750248/772870460785688576/1169680687415169024

**Comment by ambv** [comment]
When you moved the dials really quickly, 30fps was visibly not entirely smooth. 60fps works just fine but looks much better in this scenario.

OK, this is ready for review now.

**Review by tehn — APPROVED** [review]
very nice, thank you!

**Pull request merged**